app/cdap/components/ConfigurationGroup/PropertyRow/index.tsx

```diff
@@ -122,9 +126,11 @@ class PropertyRowView extends React.Component<IPropertyRowProps, IState> {
     const newValue = !this.state.isMacroTextbox;
 
     if (newValue) {
-      this.handleChange('${}');
+      this.setState({ prevValue: this.props.value });
```

Comment:
shouldn't we only set state once at the end of this? it looks like it'll run setState twice on every toggleMacro?

Reply:
Good catch, missed the setState called down below. Addressed in [1887bc9](https://github.com/cdapio/cdap-ui/pull/252/commits/1887bc960b6612b5b692dc8614104bbb50a7b35e)